Static build of armv7-rpi2-linux-gnueabihf sample fails

With crosstool-NG version 8433e99 and a defconfig modified from the rpi2 sample, I can't build a static toolchain successfully.

```
CT_ARCH_CPU="cortex-a7"
CT_ARCH_FPU="neon-vfpv4"
CT_ARCH_FLOAT_HW=y
CT_ARCH_arm=y
CT_ARCH_SUFFIX="v7"
CT_STATIC_TOOLCHAIN=y
CT_TARGET_VENDOR="rpi2"
CT_KERNEL_linux=y
CT_BINUTILS_LINKER_LD_GOLD=y
CT_BINUTILS_GOLD_THREADS=y
CT_BINUTILS_LD_WRAPPER=y
CT_CC_LANG_CXX=y
```

```
[INFO ]  Installing binutils for host
[ERROR]    g++: error: unrecognized command line option '-all-static'
[ERROR]    make[5]: *** [dwp] Error 1
[ERROR]    make[4]: *** [all-recursive] Error 1
[ERROR]    make[3]: *** [all] Error 2
[ERROR]    make[2]: *** [all-gold] Error 2
[ERROR]    make[1]: *** [all] Error 2
[ERROR]
[ERROR]  >>
[ERROR]  >>  Build failed in step 'Installing binutils for host'
[ERROR]  >>        called in step '(top-level)'
[ERROR]  >>
[ERROR]  >>  Error happened in: CT_DoExecLog[scripts/functions@257]
[ERROR]  >>        called from: do_binutils_backend[scripts/build/binutils/binut                                                                                                                                                             ils.sh@236]
[ERROR]  >>        called from: do_binutils_for_host[scripts/build/binutils/binu                                                                                                                                                             tils.sh@101]
[ERROR]  >>        called from: main[scripts/crosstool-NG.sh@638]
[ERROR]  >>
[ERROR]  >>  For more info on this error, look at the file: 'build.log'
[ERROR]  >>  There is a list of known issues, some with workarounds, in:
[ERROR]  >>      '/opt/crosstool/share/doc/crosstool-ng/8433e99/B - Known issues                                                                                                                                                             .txt'
[ERROR]
[ERROR]  (elapsed: 1:37.86)
[01:38] / make: *** [build] Error 2
```

From the log, the failing command seems to be 
`[ALL  ]    x86_64-build_unknown-linux-gnu-g++ -W -Wall    -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -frandom-seed=dwp -O2 -g -pipe   -static-libstdc++ -static-libgcc -all-static -o dwp dwp.o libgold.a ../libiberty/libiberty.a   -lpthread  -ldl`

--- a/config/binutils/binutils.in
+++ b/config/binutils/binutils.in
@@ -180,6 +180,13 @@ config BINUTILS_GOLD_SUPPORTS_ARCH
     default y if ARCH_arm
     default y if ARCH_x86
 
+config BINUTILS_GOLD_SUPPORT
+    bool
+    default y
+    depends on BINUTILS_HAS_GOLD
+    depends on BINUTILS_GOLD_SUPPORTS_ARCH
+    depends on ! STATIC_TOOLCHAIN
+
 config BINUTILS_HAS_PLUGINS
     bool
 
@@ -205,8 +212,7 @@ config BINUTILS_LINKER_LD
 config BINUTILS_LINKER_GOLD
     bool
     prompt "gold"
-    depends on BINUTILS_HAS_GOLD
-    depends on BINUTILS_GOLD_SUPPORTS_ARCH
+    depends on BINUTILS_GOLD_SUPPORT
     depends on ! BINUTILS_FORCE_LD_BFD
     select BINUTILS_GOLD_INSTALLED
     help
@@ -218,8 +224,7 @@ config BINUTILS_LINKER_GOLD
 config BINUTILS_LINKER_LD_GOLD
     bool
     prompt "ld, gold"
-    depends on BINUTILS_HAS_GOLD
-    depends on BINUTILS_GOLD_SUPPORTS_ARCH
+    depends on BINUTILS_GOLD_SUPPORT
     select BINUTILS_GOLD_INSTALLED
     select BINUTILS_LINKER_BOTH
     help
@@ -231,8 +236,7 @@ config BINUTILS_LINKER_LD_GOLD
 config BINUTILS_LINKER_GOLD_LD
     bool
     prompt "gold, ld"
-    depends on BINUTILS_HAS_GOLD
-    depends on BINUTILS_GOLD_SUPPORTS_ARCH
+    depends on BINUTILS_GOLD_SUPPORT
     select BINUTILS_GOLD_INSTALLED
     select BINUTILS_LINKER_BOTH
     select BINUTILS_LD_WRAPPER if BINUTILS_FORCE_LD_BFD

